stakingModel/contracts/FirstClaimPool.sol

```diff
@@ -0,0 +1,109 @@
+pragma solidity 0.5.4;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "@daostack/arc/contracts/controller/Avatar.sol";
+import "../../contracts/dao/schemes/FeelessScheme.sol";
+import "../../contracts/dao/schemes/ActivePeriod.sol";
+import "./UBIScheme.sol";
+
+/**
+* @title FirstClaimPool contract that transfer bonus tokens when claiming for
+* the first time
+*/
+contract FirstClaimPool is FeelessScheme, ActivePeriod {
+    using SafeMath for uint256;
+
+    UBIScheme public ubi;
+    uint256 public claimAmount;
+
+    modifier onlyUBIScheme {
+        require(
+            msg.sender == address(ubi),
+            "Only UBIScheme can call this method"
+        );
+        _;
+    }
+
+    modifier ubiHasInitialized {
+        require(address(ubi) != address(0), "ubi has not initialized");
+        _;
+    }
+
+    constructor(
+        uint256 _claimAmount,
+        Avatar _avatar,
+        Identity _identity
+    )
+        public
+        FeelessScheme(_identity, _avatar)
+        ActivePeriod(now, now * 2)
+    {
+        claimAmount = _claimAmount;
+    }
+
+    /* @dev Start function. Adds this contract to identity as a feeless scheme.
+     * Can only be called if scheme is registered
+     */
+    function start()
+        public
+        onlyRegistered
+    {
+        addRights();
+        super.start();
+    }
+
+    /**
+     * @dev sets the ubi scheme
+     * @param _ubi contract
+     */
+    function setUBIScheme(UBIScheme _ubi) public onlyAvatar {
+        ubi = _ubi;
```

Comment:
need to add to UBIScheme a call to avatar on start() to set the ubischeme

Reply:
yes we need to use a genericcall, since only avatar can set the ubischeme, then when ubisceheme starts it needs to sets itself here to have permissions to call awarduser